### PR TITLE
Validate topic name

### PR DESCRIPTION
**Pull request by patuwwy**
**What?**  
Validate topic name when using `si topic send/get`

**Why?**  
https://github.com/scramjetorg/transform-hub/issues/426

**Review checks:**

These aspects need to be checked by the reviewer:

- [ ] Verify and confirm operation (please post a screenshot) 
- [ ] All STH tests pass
- [ ] All [Scramjet Cloud Platform](https://docs.scramjet.org/platform) tests pass
- [ ] Documentation is updated or no changes

### CODE REVIEW

**Comment by BKscramjet** [comment]
Task linked: [SHN-583 Topic name validation: topic names including slashes](https://app.clickup.com/t/24308805/SHN-583)

**Comment by a-tylenda** [comment]
Error handling works just fine. I was just wondering about a more precise message regarding the characters that the user should avoid in the name. What do you think?

image
image

File: packages/cli/src/lib/commands/topic.ts [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 import { getHostClient, getReadStreamFromFile } from "../common";
 import { profileConfig } from "../config";
 import { displayEntity, displayStream } from "../output";
 
+const validateTopicName = (topicName: string) => {
+    if (topicName.match(/^[a-zA-Z0-9_-]*$/)) {
+        return;
+    }
+
+    throw new Error("Invalid topic name");
```
Comment (a-tylenda) [inline packages/cli/src/lib/commands/topic.ts:13]: Some characters are acceptable and some are not, for example, slash throws an error but backslash is ok:

image

Should we be more precise about the characters that cannot be used in the topic name? Maybe something like this:

```suggestion
    throw new Error("Invalid topic name! Please do not use special characters in the topic name, such as: / ! @ # % ^ & * ( ) + | < >");
```

**Review by a-tylenda — APPROVED** [review]

File: packages/cli/src/lib/commands/topic.ts (hunk 1 above)
Comment (patuwwy) [reply packages/cli/src/lib/commands/topic.ts:13]: actually `si topic send topic\name READM.md` creates `topicname` topic due to escape character `\`

**Comment by patuwwy** [comment]
but yes,
```
curl --url 'http://0.0.0.0:8000/api/v1/topic/topic\name' -d "data"
```
```
si topic ls
[
  {
    contentType: 'application/x-www-form-urlencoded',
    localProvider: 'api',
    topic: 'topic\\name'
  }
]
```

File: packages/cli/src/lib/commands/topic.ts (hunk 1 above)
Comment (MichalCz) [reply packages/cli/src/lib/commands/topic.ts:13]: I don't actually see anything wrong with `/` if properly escaped.

File: packages/cli/src/lib/commands/topic.ts [hunk 2]
```diff
@@ -1,9 +1,18 @@
+/* eslint-disable no-console */
 import { CommandDefinition } from "../../types";
 import { isDevelopment } from "../../utils/envs";
 import { getHostClient, getReadStreamFromFile } from "../common";
 import { profileConfig } from "../config";
 import { displayEntity, displayStream } from "../output";
 
+const validateTopicName = (topicName: string) => {
+    if (topicName.match(/^[a-zA-Z0-9_-]*$/)) {
```
Comment (MichalCz) [inline packages/cli/src/lib/commands/topic.ts:9]:
```suggestion
    if (topicName.match(/^[a-zA-Z0-9_-+,\.]*$/)) {
```

Comment (MichalCz) [reply packages/cli/src/lib/commands/topic.ts:9]: Maybe the `+` as that's commonly converted to space in `url`s...

**Comment by alicja-gruzdz** [comment]
> but yes,
> 
> ```
> curl --url 'http://0.0.0.0:8000/api/v1/topic/topic\name' -d "data"
> ```
> 
> ```
> si topic ls
> [
>   {
>     contentType: 'application/x-www-form-urlencoded',
>     localProvider: 'api',
>     topic: 'topic\\name'
>   }
> ]
> ```
> 
> we also need to validate topic names when creating topics on host

I added https://app.clickup.com/t/24308805/SHN-816

File: packages/cli/src/lib/commands/topic.ts (hunk 1 above)
Comment (alicja-gruzdz) [reply packages/cli/src/lib/commands/topic.ts:13]: I'd stay with unreserved and safe url chars, also due to user friendly looking urls.

**Review by a-tylenda — APPROVED** [review]

**Pull request merged**